.github/workflows/generate-lua-library.yml

```diff
@@ -0,0 +1,55 @@
+name: Generate Lua library
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+    paths:
+      - 'rts/Lua/**'
+jobs:
+  generate-library:
+    name: Regenerate library
+    runs-on: ubuntu-latest
+    steps:
+      - name: Configure git
+        run: |
+          git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git config --global user.name "github-actions[bot]"
+
+      - name: Checkout Lua CPP files
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: rts/Lua
+          path: recoil
+
+      # NOTE: This step is duplicated in `publish-site.yml`
+      - name: Generate Lua library
+        working-directory: recoil/rts/Lua/
+        run: |
+          npm install -g lua-doc-extractor
+          lua-doc-extractor *.cpp --dest library/generated
```

Comment:
Just as a note, it would be good to cancel the workflow if the output of any of the docgen steps errors, see: https://github.com/beyond-all-reason/spring/issues/1760 .

If comfortable doing here all good, otherwise, fine with another PR

Reply:
Yes, I believe a failure of `npm install` or `lua-doc-extractor` (non-zero exit code) will fail the workflow. However at present the only fail case for `lua-doc-extractor` is [when no source files are provided](https://github.com/rhys-vdw/lua-doc-extractor/blob/8fa5cad055b11f0990b78e517a9a0acf3ab5dc2a/src/cli.ts#L79).

We can add more error handling in future, but at the moment invalid attributes will just be copied over wholesale (or, [in some cases](https://github.com/rhys-vdw/lua-doc-extractor/issues/7) mangled).

Reply:
We might be able to do a [check with lua-language-server](https://luals.github.io/wiki/usage/#--check) on the output from `lua-doc-extractor`. I would prefer to update #1760 with this info rather than tackle it here though.

Probably the ideal would be to set a CI check on each PR that generates the docs and does a `--check` before they are merged (rather than failing the library update or site generation).